javascript-view-swapping/main.js

```diff
@@ -0,0 +1,25 @@
+var tabContainer = document.querySelector('.tab-container');
+var tabs = document.querySelectorAll('.tab');
+var view = document.querySelectorAll('.view');
+
+function tabCon(event) {
+  if (event.target.matches('.tab')) {
+    for (var i = 0; i < tabs.length; i++) {
+      tabs[i].setAttribute('class', 'tab');
+    }
+    if (!event.target.matches('.active')) {
+      event.target.setAttribute('class', 'tab active');
+    }
+  }
+
+  var viewing = event.target.getAttribute('data-view');
+  for (var j = 0; j < view.length; j++) {
+    if (view[j].getAttribute('data-view') === viewing) {
+      view[j].className = 'view';
+    } else {
+      view[j].className = 'view hidden';
+    }
+  }
```

Comment:
This should be inside your conditional to check if a tab was clicked also. As it is now, if a user clicked the tab container in an area that was not a tab, I believe your entire view would disappear 🤔